app/apollo/resolvers/group.js

```diff
@@ -507,20 +547,12 @@ const groupResolvers = {
             name: group.name,
           };
         });
-        const clusters = await commonClusterSearch(models, {org_id}, { limit: 0, skip: 0, startingAfter: null });
-        const clusterObjs = _.map(clusters, (cluster)=>{
-          return {
-            name: cluster.registration.name,
-            uuid: cluster.cluster_id,
-            registration: cluster.registration
-          };
-        });
```

Comment:
While the original code appears to be incorrectly getting **all** clusters for the pluginContext (it should be just getting a single cluster), it's incorrect to just remove the retrieval of cluster details and send just the cluster _uuid_ to the plugins -- just the uuid is insufficient for usability.  Instead the single cluster should be retrieved and that single cluster (simplified cluster with just name, uuid, and registration) added to pluginContext

Reply:
Note however that this function does not _require_ the cluster to exist (the response includes `modified: res.modifiedCount`, but that could be zero), so pluginContext could include `cluster: undefined` which is valid.

Reply:
Change coming up for this one. It will need reviewed

Reply:
Resolved

Reply:
Unresolving -- part of the problem wasn't addressed, or was reverted with ther changes.

As stated earlier, it is possible for the cluster to not be found, and `cluster` variable to be null.  As a result, it will throw an unexpected exception when attempting to do `name: cluster.registration.name` etc while generating the `clusterObjs` for plugin context.

The variable `clusterObjs` should also be made singular `clusterObj` since it is always just one.